cs464-group-project/src/pages/Home.js

```diff
@@ -1,27 +1,183 @@
+        //get only id, capacity and team name
+        const teamRanks = data.table.map((team) => ({
+          id: team.idTeam,
+          name: team.strTeam,
+          capacity: team.intStadiumCapacity,
+          rank: team.intRank,
+          wins: team.intWin,
+          loss: team.intLoss,
+          points: team.intPoints,
+          logo: team.strTeamBadge,
+        }));
+        setRanking(teamRanks);
+
+        const pastMatchesData = await getPLPastMatches(4328);
+        const tempPastMatches = pastMatchesData.events.map((match) => {
+          const homeTeam = teamRanks.find(
+            (team) => team.id === match.idHomeTeam
+          );
+          const awayTeam = teamRanks.find(
+            (team) => team.id === match.idAwayTeam
+          );
+
+          return {
+            id: match.idEvent,
+            homeId: match.idHomeTeam,
+            awayId: match.idAwayTeam,
+            homeName: match.strHomeTeam,
+            awayName: match.strAwayTeam,
+            homeScore: match.intHomeScore,
+            awayScore: match.intAwayScore,
+            eventDate: match.dateEvent,
+            homeLogo: homeTeam ? homeTeam.logo : null,
+            awayLogo: awayTeam ? awayTeam.logo : null,
+          };
+        });
+
+        const groupedMatches = tempPastMatches.reduce((acc, match) => {
+          const date = match.eventDate;
+
+          //create a property for each date storing in an array of matches
+          acc[date] = acc[date] || [];
+          acc[date].push(match);
+          return acc;
+        }, {});
+        setPastMatches(groupedMatches);
+      } catch (err) {
+        console.error('Error getting League information', err);
       }
-    }
+    };
+    fetchData();
+  }, []);
 
+  //getting live matches
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await getPLLiveScores();
+        if (data.events === null) {
+          return;
+        }
+        console.log(data);
+        //Filter out non PL leagues
+        //Check console.log for live match if you'd like to see some content, replace league id
+        //Will remove for final PR
+        const plData = data.events.filter((match) => match.idLeague === '4480');
+        console.log(data);
+        const liveMatches = plData.map((match) => ({
+          homeId: match.idHomeTeam,
+          awayId: match.idAwayTeam,
+          homeTeam: match.strHomeTeam,
+          awayTeam: match.strAwayTeam,
+          homeScore: match.intHomeScore,
+          awayScore: match.intAwayScore,
+          clock: match.strProgress,
+          homeLogo: match.strHomeTeamBadge,
+          awayLogo: match.strAwayTeamBadge,
+        }));
+        setLiveTeams(liveMatches);
+      } catch (err) {
+        console.error('Error getting League information', err);
+      }
+    };
     fetchData();
+
+    const intervalId = setInterval(() => {
+      fetchData();
+    }, 60000);
```

Comment:
You should place this in the LiveMatch Component.  Keep states that are for home page in Home component.

Reply:
So my thoughts behind doing the fetch calls here is that I want the components to be reusable.  I believe that is the main benefit of react.  Or at least that is what instructors emphasized on when I first started learning React.

So for instance if I wanted to create a live match for a different league like, Champions League, I can simply do a fetch call to get my data and then I would use that component and simply past in my data.  I don't have to create new component.
Same goes with like the `<Table ranking= {ranking} />` component.  I can simply do a fetch call for a different league and pass in another `ranking` array.  This will automatically create me a new ranking for the same league without having to write any extra code.

Let me know if you agree/disagree.

Reply:
You can reuse them by having them in their own components as well.  If I wanted to use your table rankings, I can just call your table rankings component, instead of calling your House component, which then calls your table ranking.  Hope that makes sense.  And it easier to maintain and more readable if its in its own component.